lib/common/DataStructures/BaseDictionary.h

```diff
@@ -34,6 +34,26 @@ namespace Js
 {
     template <class TDictionary>
     class RemoteDictionary;
+
+    class DefaultListLockPolicy
```

Comment:
Separate file, and may be rename?  It doesn't just lock list, but dictionary too.